timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/AutobatchingPingableLeaderFactory.java

```diff
@@ -22,36 +22,39 @@
 import java.util.Collection;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.UUID;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
 
+import org.immutables.value.Value;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.palantir.atlasdb.autobatch.Autobatchers;
 import com.palantir.atlasdb.autobatch.DisruptorAutobatcher;
-import com.palantir.common.streams.KeyedStream;
 import com.palantir.logsafe.SafeArg;
 import com.palantir.paxos.LeaderPingResult;
 import com.palantir.paxos.LeaderPingResults;
 import com.palantir.paxos.LeaderPinger;
+import com.palantir.paxos.LeaderPingerContext;
 
 public class AutobatchingPingableLeaderFactory implements Closeable {
 
     private static final Logger log = LoggerFactory.getLogger(AutobatchingPingableLeaderFactory.class);
 
     private final Collection<? extends Closeable> closeables;
     private final DisruptorAutobatcher<UUID, Optional<ClientAwarePingableLeader>> uuidToRemoteAutobatcher;
-    private final Map<? extends ClientAwarePingableLeader, ExecutorService> executors;
+    private final Map<LeaderPingerContext<BatchPingableLeader>, ExecutorService> executors;
```

Comment:
As an aside, all those executor services feel weird, can we not use retrofit and just get non-blocking futures?

Reply:
probably, it's a pretty large change, and something I have wanted to do previously, but relatively low priority